File: Problems/38_Count_and_Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        ans = "1"
        for _ in range(n-1):
            curr_num = ans[0]
            count_num = 1
            ans_next = ""
            for num in ans[1:]:
                if num == curr_num:
                    count_num += 1
                else:
                    ans_next = ans_next + str(count_num) + curr_num
                    curr_num = num
                    count_num = 1
            ans = ans_next + str(count_num) + curr_num
        return ans

if __name__ == '__main__':
    sol = Solution()
    n = 1
    print(sol.countAndSay(n))

File: Problems/38_Count_and_Say.cpp
#include<iostream>
#include<string>

using namespace std;

class Solution {
public:
    string countAndSay(int n) {
        string ans = "1";
        for(int i=1; i<n; i++){
            char curr_num = ans[0];
            int count_curr_num = 1;
            string ans_next = "";
            for(int j=1; j<ans.length(); j++){
                if (ans[j] == curr_num){
                    count_curr_num++;
                }else{
                    ans_next = ans_next + to_string(count_curr_num) + curr_num;
                    count_curr_num = 1;
                    curr_num = ans[j];
                }
            }
            ans = ans_next + to_string(count_curr_num) + curr_num;
        }
        return ans;
    }
};